src/merkle/merkle_tree.rs

```diff
@@ -114,6 +114,28 @@ impl MerkleTree {
         Ok(path.into())
     }
 
+    // ITERATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns an iterator over the leaves of this [MerkleTree].
+    pub fn leaves(&self) -> core::slice::Iter<Word> {
+        let leaves_start = self.nodes.len() / 2;
+        self.nodes[leaves_start..].iter()
+    }
```

Comment:
I wonder if we want the return type to be `impl Iterator<Item = (&u64, &Word)>` to achieve a consistent api with `SimpleSmt.leaves()`.  Potentially adding `.enumerate()` such that we have `self.nodes[leaves_start..].iter().enumerate()`.

Reply:
On second thought `.enumerate()` wont work as it returns a `usize`. We would probably want `self.nodes[leaves_start..].iter().zip(0u64..)` if we decided to go with this.

Reply:
I've tried doing this in a couple of different ways, but the issue I ran into is that we can return `(u64, &Word)` but not `(&u64, &Word)` - or, at least, I couldn't figure out how to do it without introducing too much complexity. So, I kept it as is.

Reply:
As suggested below, `&u64` should be avoided unless for very specific cases of 32-bit platforms. Normally, we just copy around these numbers; that will allow us to proceed with the suggestion from @frisitano to use `enumerate` here